controllers/storagecluster/volumegroupsnapshotterclasses.go

```diff
@@ -0,0 +1,171 @@
+}
+
+func (r *StorageClusterReconciler) createGroupSnapshotClasses(vsccs []GroupSnapshotClassConfiguration) error {
+
+	for _, vscc := range vsccs {
+		if vscc.reconcileStrategy == ReconcileStrategyIgnore || vscc.disable {
+			continue
+		}
+
+		vsc := vscc.groupSnapshotClass
+		existing := &groupsnapapi.VolumeGroupSnapshotClass{}
+		err := r.Client.Get(r.ctx, types.NamespacedName{Name: vsc.Name, Namespace: vsc.Namespace}, existing)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Since the SnapshotClass is not found, we will create a new one
+				r.Log.Info("Creating GroupSnapshotClass.", "GroupSnapshotClass", klog.KRef("", vsc.Name))
+				err = r.Client.Create(r.ctx, vsc)
+				if err != nil {
+					r.Log.Error(err, "Failed to create GroupSnapshotClass.", "GroupSnapshotClass", klog.KRef("", vsc.Name))
+					return err
+				}
+				// no error, continue with the next iteration
+				continue
+			}
+
+			r.Log.Error(err, "Failed to 'Get' GroupSnapshotClass.", "GroupSnapshotClass", klog.KRef("", vsc.Name))
+			return err
+		}
+		if vscc.reconcileStrategy == ReconcileStrategyInit {
+			return nil
+		}
+		if existing.DeletionTimestamp != nil {
+			return fmt.Errorf("failed to restore GroupSnapshotClass %q because it is marked for deletion", existing.Name)
+		}
+		// if there is a mismatch in the parameters of existing vs created resources,
+		if !reflect.DeepEqual(vsc.Parameters, existing.Parameters) {
+			// we have to update the existing SnapshotClass
+			r.Log.Info("GroupSnapshotClass needs to be updated", "GroupSnapshotClass", klog.KRef("", existing.Name))
+			existing.ObjectMeta.OwnerReferences = vsc.ObjectMeta.OwnerReferences
+			vsc.ObjectMeta = existing.ObjectMeta
+			if err := r.Client.Update(r.ctx, vsc); err != nil {
+				r.Log.Error(err, "GroupSnapshotClass updation failed.", "GroupSnapshotClass", klog.KRef("", existing.Name))
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (obj *ocsGroupSnapshotClass) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if !r.AvailableCrds[VolumeGroupSnapshotClassCrdName] {
+		r.Log.Info("VolumeGroupSnapshotClass CRD is not available")
+		return reconcile.Result{}, nil
+	}
+
+	vgsc := newGroupSnapshotClassConfigurations(instance)
+
+	err := r.createGroupSnapshotClasses(vgsc)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (obj *ocsGroupSnapshotClass) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if !r.AvailableCrds[VolumeGroupSnapshotClassCrdName] {
+		r.Log.Info("VolumeGroupSnapshotClass CRD doesn't exist")
+		return reconcile.Result{}, nil
+	}
+
+	vgscs := newGroupSnapshotClassConfigurations(instance)
+	for _, vgsc := range vgscs {
+		sc := vgsc.groupSnapshotClass
+		existing := groupsnapapi.VolumeGroupSnapshotClass{}
+		err := r.Client.Get(r.ctx, types.NamespacedName{Name: sc.Name, Namespace: sc.Namespace}, &existing)
```

Comment:
Do we really need to do a GET call? We can just make a DELETE call, and if it is not found, we can ignore it. There is no need for a GET and DELETE call.